frontend/src/components/Item/Metadata/PersonEditor.vue

```diff
@@ -46,73 +51,59 @@
         }">
         <v-spacer />
         <v-btn variant="flat" width="8em" class="mr-1" @click="handleCancel">
-          {{ $t('cancel') }}
+          {{ t('cancel') }}
         </v-btn>
-        <v-btn variant="flat" width="8em" color="primary" type="submit">
-          {{ $t('save') }}
+        <v-btn variant="flat" width="8em" color="primary" @click="handleSubmit">
+          {{ t('save') }}
         </v-btn>
       </v-card-actions>
     </v-card>
   </v-dialog>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue';
+import { useI18n } from 'vue-i18n';
 import { BaseItemPerson, ImageType } from '@jellyfin/sdk/lib/generated-client';
 
-export default defineComponent({
-  props: {
-    person: {
-      type: Object,
-      default: (): BaseItemPerson => ({
-        Name: '',
-        Type: '',
-        Role: ''
-      })
-    },
-    dialog: {
-      type: Boolean,
-      default: false
-    }
-  },
-  setup() {
-    return { ImageType };
-  },
-  data() {
-    return {
-      editState: {} as BaseItemPerson,
-      options: [
-        { text: this.$t('actor'), value: 'Actor' },
-        { text: this.$t('composer'), value: 'Composer' },
-        { text: this.$t('director'), value: 'Director' },
-        { text: this.$t('guestStar'), value: 'GuestStar' },
-        { text: this.$t('producer'), value: 'Producer' },
-        { text: this.$t('writer'), value: 'Writer' }
-      ]
-    };
-  },
-  watch: {
-    person(value: BaseItemPerson): void {
-      if (!value) {
-        return;
-      }
+const props = defineProps<{ person: BaseItemPerson | undefined }>();
+
+const emit = defineEmits<{
+  (e: 'update:person', person: BaseItemPerson): void;
+  (e: 'close'): void;
+}>();
+
+const { t } = useI18n();
 
-      this.editState = value;
-    }
+const editState = ref<BaseItemPerson>();
+const options = computed(() => [
+  { text: t('actor'), value: 'Actor' },
+  { text: t('composer'), value: 'Composer' },
+  { text: t('director'), value: 'Director' },
+  { text: t('guestStar'), value: 'GuestStar' },
+  { text: t('producer'), value: 'Producer' },
+  { text: t('writer'), value: 'Writer' }
```

Comment:
These adjust the `Type` field on `BaseItemPerson`, which is defined as `string`. Form the Jellyfin server code, it's [defined as a string](https://github.com/jellyfin/jellyfin/blob/f3c57e6a0ae015dc51cf548a0380d1bed33959c2/MediaBrowser.Model/Dto/BaseItemPerson.cs#L36). However, there is a [PersonType.cs](https://github.com/jellyfin/jellyfin/blob/f3c57e6a0ae015dc51cf548a0380d1bed33959c2/MediaBrowser.Model/Entities/PersonType.cs) class which seems to be an enumeration of all the possible values, but for some reason it's not an enum (might be a fun first bug for working on the server code :D). Since it's not an enum, the TS SDK types aren't picked up for it though.

TL;DR It's not a `BaseItemKind` and we don't have complete types for it :(

Reply:
BaseItemKind didn't exist before either, it was added by @crobibero. There's something wonky in those enums that it's not caught up by the OpenAPI server generator.

Maybe he can give you some insights in case you want to dive in server land :P

Reply:
I looked at making PersonType an enum but it can be complete customized by plugins.. so while it is possible we would need to also migrate the database and make a lot of server code changes

Reply:
Oof that's unfortunate that's it could be anything :( I'll leave this as is for now